### PR TITLE
[SYSTEMML-1419] Cleanup nested if-elses in main algorithms

### PR DESCRIPTION
**Pull request by j143**
Only 1. `GLM.dml` , 2. `GLM-predict.dml` have been changed and all the other scripts are OK.
 - changed the `if ( ) { else { if ( ) else ..}}` to `if () { } else if () { } ... else { }`

### CODE REVIEW

**Comment by akchinSTC** [comment]
Can an Admin verify this patch?

**Comment by j143** [comment]
@akchinSTC  can you test this?

**Comment by j143** [comment]
ping @deroneriksson

**Comment by deroneriksson** [comment]
Test this Jenkins

**Comment by deroneriksson** [comment]
add to whitelist.

**Comment by deroneriksson** [comment]
Thank you @j143! Would you mind adding a comment to [SYSTEMML-1419](https://issues.apache.org/jira/browse/SYSTEMML-1419) saying that you'd like to work on this issue? That way, I can add your JIRA userid as a SystemML contributor and assign the JIRA to you. Thanks!

**Comment by deroneriksson** [comment]
@j143 Nevermind regarding the JIRA comment. I found your JIRA userid on [SYSTEMML-1437](https://issues.apache.org/jira/browse/SYSTEMML-1437) so I was able to add you as a contributor and assign the JIRA to you.

**Comment by j143** [comment]
I think only the above two files contains the old `if ( ) else { }` style and have been fixed, and I've gone through all the algorithms.

**Comment by j143** [comment]
@deroneriksson can you assign me to [systemml-888](https://issues.apache.org/jira/browse/SYSTEMML-888), until my systemml-1437 is reviewed and tested, I will implement the algorithm.

**Comment by deroneriksson** [comment]
Great @j143, I've assigned SYSTEMML-888 to you. Note in the comment that I supplied that there is a PNMF example at http://apache.github.io/systemml/spark-mlcontext-programming-guide.html#define-a-kernel-for-poisson-nonnegative-matrix-factorization-pnmf-in-dml that is probably the best starting point for the creation of a PNMF.dml file.

**Comment by deroneriksson** [comment]
I believe there may be some more nested if-else's that need to be cleaned up. Could you please check on the following?
```
GLM-predict.dml
Line 108
Line 112
Line 193

GLM.dml
Line 254
Line 640
Line 787
Line 885
Line 958
```

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1720/

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1729/

**Comment by j143** [comment]
interesting! @deroneriksson 
Passed all GLM related tests. Failed with:
```00:16:55 Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 83.021 sec <<< FAILURE! - in org.apache.sysml.test.integration.applications.pydml.LinearLogRegPyDMLTest
00:16:55 testLinearLogRegPyDml[3](org.apache.sysml.test.integration.applications.pydml.LinearLogRegPyDMLTest)  Time elapsed: 35.7 sec  <<< FAILURE!
00:16:55 java.lang.AssertionError: Detailed matrices characteristics:
00:16:55 !  wR<->wSYSTEMML # stored values in wSYSTEMML: 1000
00:16:55 !  wR<->wSYSTEMML # stored values in wR: 1000
00:16:55 !  wR<->wSYSTEMML identical values(z=0): 0
00:16:55 !  wR<->wSYSTEMML wrong values(z=1.0E-14): 1000
00:16:55 !  wR<->wSYSTEMML min error: 1.347879932667162E-10
00:16:55 !  wR<->wSYSTEMML max error: 3.591591273587255E-4
```

**Comment by j143** [comment]
@deroneriksson can you review this again? thanks.

**Comment by deroneriksson** [comment]
Sure, I will have a look today.

**Comment by deroneriksson** [comment]
Thank you for working on this @j143. GLM-predict.dml LGTM. I will need another day to review GLM.dml. The previous code (which has things such as 22 or 23 levels of nested if/else statements) is quite tricky to review and validate.

File: scripts/algorithms/GLM.dml [hunk 1]
```diff
@@ -519,47 +519,47 @@ check_if_supported =
...
-    }   }}}}} }}}
+        if (link_type == 1 & link_power == -1.0)      print ("Binomial.inverse");     
+        else if (link_type == 1 & link_power ==    0) print ("Binomial.log");          
+        else if (link_type == 1 & link_power ==  0.5) print ("Binomial.sqrt");         
+        else if (link_type == 1 & link_power ==  1.0) print ("Binomial.id");           
+        else if (link_type == 1)                      print ("Binomial.power_nonlog"); 
+        else if (link_type == 2)                      print ("Binomial.logit");        
+        else if (link_type == 3)                      print ("Binomial.probit");       
+        else if (link_type == 4)                      print ("Binomial.cloglog");      
+        else (link_type == 5)                         print ("Binomial.cauchit");
```
Comment (deroneriksson) [inline scripts/algorithms/GLM.dml]: i believe L561 `else` needs to be `else if`

**Comment by akchinSTC** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://sparktc.ibmcloud.com/jenkins/job/SystemML-PullRequestBuilder/1758/

**Comment by j143** [comment]
I have gone through the code freshly again. It looks fine to me.

**Comment by deroneriksson** [comment]
LGTM. Thank you @j143!

**Comment by j143** [comment]
Hi @deroneriksson , this script (https://github.com/apache/systemml/blob/912c65506d626c8b0128ceb80744fde49efd4a1a/src/test/scripts/applications/glm/GLM.dml#L23) needs, if else clean up. This is nothing but the copy of GLM.dml . Please consider changing this file.

**Pull request closed without merging**